sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/TextAnalyticsActions.java

```diff
@@ -145,9 +160,19 @@ public TextAnalyticsActions setExtractKeyPhrasesActions(ExtractKeyPhrasesAction.
      * @param analyzeSentimentActions the list of {@link AnalyzeSentimentAction} to be executed.
      *
      * @return the {@link TextAnalyticsActions} object itself.
+     *
+     * @throws IllegalArgumentException if duplicate {@link AnalyzeSentimentAction} actions are passed.
      */
     public TextAnalyticsActions setAnalyzeSentimentActions(AnalyzeSentimentAction... analyzeSentimentActions) {
+        validateActionsNumber(analyzeSentimentActions, AnalyzeSentimentAction.class.getName());
         this.analyzeSentimentActions = analyzeSentimentActions == null ? null : Arrays.asList(analyzeSentimentActions);
         return this;
     }
+
+    private void validateActionsNumber(Object[] actions, String actionType) {
```

Comment:
What happens if we do not check it in the client code?
Does the service throw an error in that case?

Reply:
@samvaity Please refer to https://github.com/Azure/azure-sdk-for-js/issues/16128 for more info on why this is needed.